src/com/sap/piper/integration/TransportManagementService.groovy

```diff
@@ -0,0 +1,141 @@
+
+    }
+
+
+    def uploadFile(String url, String token, String file, String namedUser) {
+
+        echo("File upload started.")
+
+        if (config.verbose) {
+            echo("URL: '${url}', File: '${file}'")
+        }
+
+        def httpResponse = script.sh returnStdout: true,
+            script: """#!/bin/sh -e
+                        curl -H 'Authorization: Bearer ${token}' -F 'file=@${file}' -F 'namedUser=${namedUser}' -o responseFileUpload.txt --write-out '%{http_code}' --fail '${url}/v2/files/upload'
+                    """
+
+        if (httpResponse.toInteger() < 200 || httpResponse.toInteger() >= 300) {
+            script.error "[TransportManagementService] File upload failed. HTTP-Status: ${httpResponse}"
+        }
+
+        def responseContent = script.readFile("responseFileUpload.txt")
+
+        if (config.verbose) {
+            echo("${responseContent}")
+        }
+
+        echo("File upload successful.")
+
+        return jsonUtils.jsonStringToGroovyObject(responseContent)
+
+    }
+
+
+    def uploadFileToNode(String url, String token, String nodeName, int fileId, String description, String namedUser) {
+
+        echo("Node upload started.")
+
+        if (config.verbose) {
+            echo("URL: '${url}', NodeName: '${nodeName}', FileId: '${fileId}''")
+        }
+
+        def bodyMap = [nodeName: nodeName, contentType: 'MTA', description: description, storageType: 'FILE', namedUser: namedUser, entries: [[uri: fileId]]]
+
+        def parameters = [
+            url          : "${url}/v2/nodes/upload",
+            httpMode     : "POST",
+            contentType  : 'APPLICATION_JSON',
+            requestBody  : jsonUtils.groovyObjectToPrettyJsonString(bodyMap),
+            customHeaders: [
+                [
+                    maskValue: true,
+                    name     : 'authorization',
+                    value    : "Bearer ${token}"
+                ]
+            ]
+        ]
+
+        def response = sendApiRequest(parameters)
+        echo("Node upload successful.")
+
+        return jsonUtils.jsonStringToGroovyObject(response)
+
+    }
+
+    private sendApiRequest(parameters) {
+        def defaultParameters = [
+            acceptType            : 'APPLICATION_JSON',
+            quiet                 : !config.verbose,
+            consoleLogResponseBody: !config.verbose,
+            ignoreSslErrors       : true,
+            validResponseCodes    : "100:399"
+        ]
+
+        def response = script.httpRequest(defaultParameters + parameters)
+
+        if (config.verbose) {
+            echo("Received response '${response.content}' with status ${response.status}.")
+        }
+
+        return response.content
+    }
+
+    private echo(message) {
+        script.echo "[${getClass().getSimpleName()}] ${message}"
+    }
+
+    private static String urlEncodeAndReplaceSpace(String data) {
+        return URLEncoder.encode(data, "UTF-8").replace('%20', '+')
```

Comment:
Out of curiosity: what is the reason for replaced encoded space (`%20`) by `+` signs?

Reply:
From https://www.w3.org/TR/html401/interact/forms.html#h-17.13.3.3
> **application/x-www-form-urlencoded**  
This is the default content type. Forms submitted with this content type must be encoded as follows:
>1. Control names and values are escaped. Space characters are replaced by `+', and then reserved characters are escaped as described in [RFC1738], section 2.2: Non-alphanumeric characters are replaced by `%HH', a percent sign and two hexadecimal digits representing the ASCII code of the character. Line breaks are represented as "CR LF" pairs (i.e., `%0D%0A').
>1. The control names/values are listed in the order they appear in the document. The name is separated from the value by `=' and name/value pairs are separated from each other by `&'.